editor/src/clj/editor/collision_object.clj

```diff
@@ -409,12 +411,61 @@
+                                            [(mapv min min-point point) (mapv max max-point point)])
+                                          [(repeat Double/MAX_VALUE) (repeat Double/MIN_VALUE)]
+                                          points)
+          aabb (geom/coords->aabb max-coords min-coords)
+          center (-> aabb geom/aabb-center types/Point3d->Vec3)
+          point-count (count points)
+          pos-vtx-put! (fn [vb [x y z]] (scene-shapes/pos-vtx-put! vb x y z 0.0))
+          triangles (->> points
+                         (reduce-kv (fn [vb i point]
+                                      (let [next-point (nth points (inc i) (first points))]
+                                        (reduce pos-vtx-put! vb [center point next-point])))
+                                    (scene-shapes/->pos-vtx (* 3 point-count) :static))
+                         vtx/flip!)
+          lines (->> points
+                     (reduce-kv (fn [vb i point]
+                                  (let [next-point (nth points (inc i) (first points))]
+                                    (reduce pos-vtx-put! vb [point next-point])))
+                                (scene-shapes/->pos-vtx (* 2 point-count) :static))
+                     vtx/flip!)]
+      {:node-id _node-id
+       :node-outline-key "Convex Hull"
+       :aabb aabb
+       :renderable {:render-fn render-triangles-uniform-scale
+                    :tags #{:collision-shape}
+                    :passes [pass/transparent pass/selection]
+                    :user-data {:color color
+                                :double-sided true
+                                :geometry {:primitive-type GL2/GL_TRIANGLES
+                                           :vbuf triangles}}}
+       :children [{:node-id _node-id
+                   :aabb aabb
+                   :renderable {:render-fn render-lines-uniform-scale
+                                :tags #{:collision-shape :outline}
+                                :passes [pass/outline]
+                                :user-data {:color color
+                                            :geometry {:primitive-type GL2/GL_LINES
+                                                       :vbuf lines}}}}]})))
+
 (g/defnk produce-scene
-  [_node-id child-scenes]
-  {:node-id _node-id
-   :aabb geom/null-aabb
-   :renderable {:passes [pass/selection]}
-   :children child-scenes})
+  [_node-id child-scenes collision-shape dep-build-targets collision-group-color]
+  (let [convex-shape-data (when (and collision-shape (= (resource/type-ext collision-shape) "convexshape"))
+                            (-> dep-build-targets ffirst :user-data :pb))]
+    {:node-id _node-id
+     :aabb geom/null-aabb
+     :renderable {:passes [pass/selection]}
+     :children (if convex-shape-data
+                 [(convex-hull-scene _node-id convex-shape-data collision-group-color)]
+                 child-scenes)}))
```

Comment:
If there is a convex shape resource set, the supported built-in shapes cannot be used as far as I understand.